sycl/test/on-device/online_compiler/online_compiler_L0.cpp

```diff
@@ -0,0 +1,49 @@
+// REQUIRES: level_zero
+
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple -I %sycl_source_dir -lze_loader %s -o %t.out
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+
+// This test checks INTEL feature class online_compiler for Level-Zero.
+// All Level-Zero specific code is kept here and the common part that can be
+// re-used by other backends is kept in online_compiler_common.hpp file.
+
+#include <CL/sycl.hpp>
+#include <CL/sycl/INTEL/online_compiler.hpp>
+
+#include <vector>
+
+// clang-format off
+#include <level_zero/ze_api.h>
+#include <CL/sycl/backend/level_zero.hpp>
+// clang-format on
+
+using byte = unsigned char;
+
+sycl::kernel getSYCLKernelWithIL(sycl::context &Context,
+                                 const std::vector<byte> &IL) {
+
+  ze_module_desc_t ZeModuleDesc = {};
+  ZeModuleDesc.format = ZE_MODULE_FORMAT_IL_SPIRV;
+  ZeModuleDesc.inputSize = IL.size();
+  ZeModuleDesc.pInputModule = IL.data();
+  ZeModuleDesc.pBuildFlags = "";
+  ZeModuleDesc.pConstants = nullptr;
+
+  assert(Context.get_devices().size() == 1 && "Expected to have only 1 device");
+  sycl::device Device = Context.get_devices()[0];
```

Comment:
What is the plan for context with multiple devices? I think we need to build (`ZeModuleBuild` to all devices in the context).

Reply:
I am not sure I understand why we need to build for all devices in the context.
The online_compiler SPEC defines the constructor with purpose to compile only for one device.
SPEC owner @kbobrovs , please advise how to proceed.

Reply:
To compile for multiple devices, user program will obtain device list from the SYCL context and invoke the online_compiler::compile for each.